src/lv_misc/lv_mem.c

```diff
@@ -325,7 +312,7 @@ void * lv_mem_realloc(void * data_p, size_t new_size)
 
     if(data_p != NULL) {
         /*Copy the old data to the new. Use the smaller size*/
-        if(old_size != 0) {
+        if(old_size != 0 && new_size != 0) {
             _lv_memcpy(new_p, data_p, LV_MATH_MIN(new_size, old_size));
             lv_mem_free(data_p);
```

Comment:
if `new_size == 0` ` lv_mem_free(data_p);` still should be called.

Reply:
Yes, you are right. I will upate the patch.